Mathlib/NumberTheory/LSeries/SumCoeff.lean

```diff
@@ -0,0 +1,154 @@
+  refine hO.congr' (Eventually.of_forall fun _ ↦ Finset.sum_congr rfl fun _ h ↦ ?_) EventuallyEq.rfl
+  rw [if_neg (zero_lt_one.trans_le (mem_Icc.mp h).1).ne']
+
+/-- If `f` takes nonnegative real values and the partial sums `∑ k ∈ Icc 1 n, f k` are `O(n ^ r)`
+for some real `0 ≤ r`, then the L-series `LSeries f` converges at `s : ℂ` for all `s`
+such that `r < s.re`. -/
+theorem LSeriesSummable_of_sum_norm_bigO_and_nonneg
+    {f : ℕ → ℝ} (hO : (fun n ↦ ∑ k ∈ Icc 1 n, f k) =O[atTop] fun n ↦ (n : ℝ) ^ r)
+    (hf : ∀ n, 0 ≤ f n) (hr : 0 ≤ r) (hs : r < s.re) :
+    LSeriesSummable (fun n ↦ f n) s :=
+  LSeriesSummable_of_sum_norm_bigO (by simpa [_root_.abs_of_nonneg (hf _)]) hr hs
+
+end summable
+
+section integralrepresentation
+
+private theorem LSeries_eq_mul_integral_aux {f : ℕ → ℂ} (hf : f 0 = 0) {r : ℝ} (hr : 0 ≤ r) {s : ℂ}
+    (hs : r < s.re) (hS : LSeriesSummable f s)
+    (hO : (fun n ↦ ∑ k ∈ Icc 1 n, f k) =O[atTop] fun n ↦ (n : ℝ) ^ r) :
+    LSeries f s = s * ∫ t in Set.Ioi (1 : ℝ), (∑ k ∈ Icc 1 ⌊t⌋₊, f k) * t ^ (- (s + 1)) := by
+  have h₁ : (-s - 1).re + r < -1 := by
+    rwa [sub_re, one_re, neg_re, neg_sub_left, neg_add_lt_iff_lt_add, add_neg_cancel_comm]
+  have h₂ : s ≠ 0 := ne_zero_of_re_pos (hr.trans_lt hs)
+  have h₃ (t : ℝ) (ht : t ∈ Set.Ici 1) : DifferentiableAt ℝ (fun x : ℝ ↦ (x : ℂ) ^ (-s)) t :=
+    differentiableAt_id.ofReal_cpow_const (zero_lt_one.trans_le ht).ne' (neg_ne_zero.mpr h₂)
+  have h₄ : ∀ n, ∑ k ∈ Icc 0 n, f k = ∑ k ∈ Icc 1 n, f k := fun n ↦ by
+    rw [← Nat.Icc_insert_succ_left n.zero_le, sum_insert (by aesop), hf, zero_add, zero_add]
+  simp_rw [← h₄] at hO
+  rw [← integral_mul_left]
+  refine tendsto_nhds_unique ((tendsto_add_atTop_iff_nat 1).mpr hS.hasSum.tendsto_sum_nat) ?_
+  simp_rw [Nat.range_succ_eq_Icc_zero, LSeries.term_def₀ hf, mul_comm (f _)]
+  convert tendsto_sum_mul_atTop_nhds_one_sub_integral₀ (f := fun x ↦ (x : ℂ) ^ (-s)) (l := 0)
+    ?_ hf h₃ ?_ ?_ ?_ (integrableAtFilter_rpow_atTop_iff.mpr h₁)
+  · rw [zero_sub, ← integral_neg]
+    refine setIntegral_congr_fun measurableSet_Ioi fun t ht ↦ ?_
+    rw [deriv_ofReal_cpow_const (zero_lt_one.trans ht).ne', h₄]
+    · ring_nf
+    · exact neg_ne_zero.mpr <| ne_zero_of_re_pos (hr.trans_lt hs)
+  · refine integrableOn_Ici_iff_integrableOn_Ioi.mpr <|
+      integrableOn_Ioi_deriv_ofReal_cpow zero_lt_one (by simpa using hr.trans_lt hs)
+  · have hlim : Tendsto (fun n : ℕ ↦ (n : ℝ) ^ (-(s.re - r))) atTop (𝓝 0) :=
+      (tendsto_rpow_neg_atTop (by rwa [sub_pos])).comp tendsto_natCast_atTop_atTop
+    refine (IsBigO.mul_atTop_rpow_natCast_of_isBigO_rpow (-s.re) _ _ ?_ hO ?_).trans_tendsto hlim
+    · exact isBigO_norm_left.mp <| (norm_ofReal_cpow_eventually_eq_atTop _).isBigO.natCast_atTop
+    · linarith
+  · refine .mul_atTop_rpow_of_isBigO_rpow (-(s + 1).re) r _ ?_ ?_ (by rw [← neg_re, neg_add'])
+    · simpa only [add_re, one_re, neg_add'] using isBigO_deriv_ofReal_cpow_const_atTop _
```

Comment:
`simpa [-neg_add_rev, neg_add'] using isBigO_deriv_ofReal_cpow_const_atTop _` also works here, cf. [Zulip discussion on (not) squeezing terminal simps](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/request.3A.20discourage.20squeezing.20terminal.20simps.3F)

Reply:
I tried to "unsqueeze" this `simp` but it didn't work, I see that the trick was to remove `neg_add_rev`. Thanks!